pkg/matcher/matcher.go

```diff
@@ -0,0 +1,59 @@
+package matcher
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+)
+
+const (
+	methodString        = "string"
+	methodRegexp        = "regexp"
+	methodSimplePattern = "simplepattern"
+)
+
+// Matcher is an interface that wraps Match method.
+type Matcher interface {
+	Match(string) bool
+}
+
+// CreateMatcher creates matcher.
+// Syntax: method=expression, valid methods: string, regexp, simplepattern.
```

Comment:
> Can't describe a string match which starts with ^ or ends with $

Why do you need this? Choose any of these.

> simplepattern is wildcard + alpha, we need to doc it well since it's not standard syntax.

i will replace simplepattern with shell (filepath.Match).

[simplepattern](https://docs.netdata.cloud/libnetdata/simple_pattern/) btw

Reply:
> = usually means assign or equal, but here means separating

what's your suggestion?

Reply:
> ex. how to describe a string equals ^hello or hello$

^^hello or hello$$